### PR TITLE
cellSysCacheClear/Mount() improved

### PR DESCRIPTION
**Pull request by elad335**
* Clear() error checking simplified a bit
* Mount() now clears cache if ID was changed from last or null term at start specified. (hw tested)
* Clear() now uses vfs::host::remove_all() to fix behaviour on Windows.
* Added mutex for Mount().

Previous pr got closed because of unitentionally deleting branch..

### CODE REVIEW

**Comment by elad335** [comment]
I've published the testcase used at https://github.com/elad335/myps3tests/tree/master/ppu_tests/CellSysCacheMount because why not.

**Comment by rajkosto** [comment]
This breaks ability to resume playing Metal Gear Solid 4 after playing the MGS1 section:

This is what happens:
1. MGS4 creates file `/dev_hdd1/cache/s/dev_hdd1/cache/s/gclvar.bak` (after failing once, it mkdirs and tries again then succeeds)
2. MGS4 launches MGS1 self (mgs/stage/module/en.self) as new program
3. You play through MGS1 self (created gclvar.bak file is still there and accessible)
4. MGS1 launches MGS4.self back again (with some args telling it to use gclvar.bak)
5. ---- /dev_hdd1/cache/ FOLDER IS CLEARED ---- (why ? we never switched games, only exes)
6. MGS4 fails to open `/dev_hdd1/cache/s/dev_hdd1/cache/s/gclvar.bak` (fs_open returning CELL_ENOENT) because cache was cleared, and proceeds to play as if you launched it normally (unable to progress).

Here's a save file that when loaded in MGS4 immediately goes into MGS1: https://files.sshnuke.net/BLUS30109G5DA88B89.zip just crawl into the vent or die and exit to finish that section

**Comment by elad335** [comment]
Fixed.
@legend800 can you retest WRC 2 and WRC 3 to see if it's still fixed?

**Comment by legend800** [comment]
WRC2/3 still work fine after changes.

**Comment by Emulator-Team-2** [comment]
Fixes https://github.com/RPCS3/rpcs3/issues/6526.

File: rpcs3/Emu/Cell/Modules/cellSysutil.cpp [hunk 1]
```diff
@@ -72,8 +72,18 @@ extern void sysutil_send_system_cmd(u64 status, u64 param)
...
+	atomic_t<bool> mounted = false;
+	std::string cache_id;
+	shared_mutex mtx;
+
+	~syscache()
+	{
+		// Check if mounted and cache_id is different than already written empty value
+		if (!cache_id.empty())
+		{
+			fs::write_file(fs::get_cache_dir() + "/cache/cacheId", fs::rewrite, cache_id);
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellSysutil.cpp]: Maybe theres a better alternative to the way its storing last cache id from the previous boot?

**Pull request merged**